components/email.php

```diff
@@ -0,0 +1,219 @@
+	 *
+	 * [--plain-text-content=<plain-text-content>]
+	 * : Plain-text email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * [<file>]
+	 * : Read content from <file>. If this value is present, the
+	 *     `--content` argument will be ignored.
+	 *
+	 *   Passing `-` as the filename will cause post content to
+	 *   be read from STDIN.
+	 *
+	 * [--edit]
+	 * : Immediately open system's editor to write or edit email content.
+	 *
+	 *   If content is read from a file, from STDIN, or from the `--content`
+	 *   argument, that text will be loaded into the editor.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Create email post
+	 *     $ wp bp email create --type=new-event --type-description="Send an email when a new event is created" --subject="[{{{site.name}}}] A new event was created" --content="<a href='{{{some.custom-token-url}}}'></a>A new event</a> was created" --plain-text-content="A new event was created"
+	 *     Success: Email post created for type "new-event".
+	 *
+	 *     # Create email post with content from given file
+	 *     $ wp bp email create ./email-content.txt --type=new-event --type-description="Send an email when a new event is created" --subject="[{{{site.name}}}] A new event was created" --plain-text-content="A new event was created"
+	 *     Success: Email post created for type "new-event".
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$switched = false;
+
+		if ( false === bp_is_root_blog() ) {
+			$switched = true;
+			switch_to_blog( bp_get_root_blog_id() );
+		}
+
+		$term = term_exists( $assoc_args['type'], bp_get_email_tax_type() );
+
+		// Term already exists so don't do anything.
+		if ( $term !== 0 && $term !== null ) {
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::error( "Email type '{$assoc_args['type']}' already exists." );
+		}
+
+		if ( ! empty( $args[0] ) ) {
+			$assoc_args['content'] = $this->read_from_file_or_stdin( $args[0] );
+		}
+		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'edit' ) ) {
+			$input = \WP_CLI\Utils\get_flag_value( $assoc_args, 'content', '' );
+			if ( $output = $this->_edit( $input, 'WP-CLI: New BP Email Content' ) ) {
+				$assoc_args['content'] = $output;
+			} else {
+				$assoc_args['content'] = $input;
+			}
+		}
+
+		$id = $assoc_args['type'];
+
+		$defaults = array(
+			'post_status' => 'publish',
+			'post_type'   => bp_get_email_post_type(),
+		);
+
+		$email = array(
+			'post_title'   => $assoc_args['subject'],
+			'post_content' => $assoc_args['content'],
+			'post_excerpt' => ! empty( $assoc_args['plain-text-content'] ) ? $assoc_args['plain-text-content'] :'',
+		);
+
+		// Email post content.
+		$post_id = wp_insert_post( bp_parse_args( $email, $defaults, 'install_email_' . $id ), true );
+
+		// Save the situation.
+		if ( ! is_wp_error( $post_id ) ) {
```

Comment:
`wp_insert_post` can return `false`. If that happens, for some reason, this, `$post_id->get_error_message()`, will output an underfined error instead of the error message.

Reply:
Hi, sorry for the late reply.  This should be addressed in commit ec33c16.

See the second argument for `wp_insert_post()`, which forces `wp_insert_post()` to return a `WP_Error` object.